docs/AdvancedUsage.md

```diff
@@ -0,0 +1,172 @@
+| **formatDetails**: String      | ""            | Optional format details. Required if the recording cannot be interpreted with default format details. See "Supported file formats" in ecu.test help for more information.                                                                                                            |
+
+## CheckPackageResult
+
+All given properties can be read via a getter method. (e.g. `.getResult()`)
+
+| Properties                                | Description                                           |
+|-------------------------------------------|-------------------------------------------------------|
+| **result**: String                        | Result of the package check execution                 |
+| **testCasePath**: String                  | Package path of the check package                     |
+| **issues**: List\<HashMap\<String,String> | List of problems found in the package check execution |
+
+## GenerationResult
+
+All given properties can be read via a getter method. (e.g. `.getGenerationResult()`)
+
+| Properties                    | Description                               |
+|-------------------------------|-------------------------------------------|
+| **generationResult**: String  | Result of the report generation execution |
+| **generationMessage**: String | ecu.test report generation message        |
+| **reportOutputDir**: String   | location of ecu.test report               |
+
+## TestResult
+
+All given properties can be read via a getter method. (e.g. `.getReportId()`)
+
+| Properties             | Description                               |
+|------------------------|-------------------------------------------|
+| **reportId**: String   | reportId given by ecu.test                |
+| **testResult**: String | Result of the package / project execution |
+| **reportDir**: String  | location of the report                    |
+
+## UploadResult
+
+All given properties can be read via a getter method. (e.g. `.getUploadResult()`)
+
+| Properties                | Description                                |
+|---------------------------|--------------------------------------------|
+| **uploadResult**: String  | Result of the report upload to test guide. |
+| **uploadMessage**: String | Response message from test.guide           |
+| **reportLink**: String    | test guide link to the report              |
+
+## ScanMode
+
+| Properties        | Possible Values                                                         |
+|-------------------|-------------------------------------------------------------------------|
+| enum **ScanMode** | PACKAGES_ONLY <br/><br/> PROJECTS_ONLY <br/><br/> PACKAGES_AND_PROJECTS |
```

Comment:
* Describe the behavior
* in case of a pipeline definition enums will be set as strings